src/main/java/org/ms/zk/plugin/matcher/MatcherFactory.java

```diff
@@ -0,0 +1,30 @@
+package org.ms.zk.plugin.matcher;
+
+import org.ms.zk.plugin.matcher.HostMatcher;
+import org.ms.zk.plugin.matcher.Matcher;
+import org.ms.zk.plugin.matcher.UserMatcher;
+import org.ms.zk.plugin.matcher.role.AdminRoleMatcher;
+import org.ms.zk.plugin.matcher.role.ReaderRoleMatcher;
+import org.ms.zk.plugin.matcher.role.ServerRoleMatcher;
+
+import java.io.IOException;
+
+public class MatcherFactory {
+
+  public static Matcher get(String aclExpr) throws IOException {
+    if (aclExpr.startsWith("host/")) return new HostMatcher();
+
+    if (aclExpr.startsWith("role/")) {
```

Comment:
nit - isn;t it easier:

if aclExpr.startswith("role/admin") return AdminRoleMatcher() ? saving on replace/switch.